src/FormControl.js

```diff
@@ -46,6 +47,7 @@ class FormControl extends React.Component {
       id = controlId,
       inputRef,
       className,
+      bsSize,
```

Comment:
I think it's correct from a user's perspective, and wrong from ours :P 

the className mismatch is a bug/implementation detail on bootstraps parts not an intentional "different thing" like validation state sort of is. my evidence: it's fixed in v4 https://v4-alpha.getbootstrap.com/components/forms/#control-sizing